src/sentry/static/sentry/app/components/group/externalIssueActions.tsx

```diff
@@ -37,87 +37,146 @@ class ExternalIssueActions extends AsyncComponent<Props, State> {
     this.state = {
       showModal: false,
       action: 'create',
-      selectedIntegration: this.props.integration,
-      issue: this.getIssue(),
+      selectedIntegration: null,
       ...this.getDefaultState(),
+      ...this.linkedIssuesFilter(),
     };
   }
 
   getEndpoints() {
     return [];
   }
 
-  getIssue() {
-    return this.props.integration && this.props.integration.externalIssues
-      ? this.props.integration.externalIssues[0]
-      : null;
+  linkedIssuesFilter() {
+    return this.props.configurations.reduce(
+      (acc, curr) => {
+        if (curr.externalIssues.length) {
+          acc.linked.push(curr);
+        } else {
+          acc.unlinked.push(curr);
+        }
+        return acc;
+      },
+      {linked: [] as GroupIntegration[], unlinked: [] as GroupIntegration[]}
```

Comment:
@NisanthanNanthakumar why do we need typecasting here? Can we just type the `acc` value?